src/LoginDialog/OidcUtilities.ts

```diff
@@ -1,29 +1,79 @@
 const generateStateToken = () => {
-    var array = new Uint32Array(160);
+    var array = new Uint8Array(20);
     const randomValues = window.crypto.getRandomValues(array)
     return randomValues.join('');
 }
 
+
+//////////////////////////////////////////////////////////////////////
+// Proof Key Code Exchange (PKCE) HELPER FUNCTIONS
+// Adapted from: https://github.com/aaronpk/pkce-vanilla-js/blob/master/index.html#L158
```

Comment:
I'm not too fond of copy/pasting code for our Ref Exps like this. Was there a library we can pull that can implement this for us? 

Looking around, I also see the following options that can also trim down this code quite a bit:

https://github.com/brix/crypto-js

https://www.websparrow.org/web/javascript-base64-and-url-encoding-decoding-example

Reply:
> I'm not too fond of copy/pasting code for our Ref Exps like this. The code doesn't appear to be doing anything too complicated, so it looks like this can be re-written and cleaned up even further.

The front end team needs to have consistent guidance with this, that was my original approach and @dusanradovanovic  said I should redo it, without that. So I did, I am not a fan of redoing it the other way now that he is on vacation.

Additionally you should be advised, this is what the dashboard is using, https://github.com/moltin/dashboard/pull/879/files#diff-879141a0bcbf507bff065b3cac523ddc26959e27c43af5c3b42b7b7130722983R13 , except that in the dashboards case, we are using security code dickson wrote including a rejection sampling method. Arguably using code from someone in the OAuth community, is better than code we wrote ourselves.

>Was there a library we can pull that can implement this for us?
Yes kind of, except that if you remove the comments, the code that is being copy and pasted is only 21 lines. Dusan's concern was the size. https://www.npmjs.com/package/crypto-js , is 450 KB.

Reply:
FYI on this @dusanradovanovic

Reply:
Oh so I looked at this a bit more, and the code is MIT licensed, so probably should put it in it's own file or something.

Reply:
Hey @steve-r-west, researched this a bit more. I'm good with us keeping this code so long as we put it into a separate file and add the copyright header to it. From what I'm told, we can do whatever we want with the code under MIT license but we need to add a copy of the original MIT license and copyright notice to it.